torch/tensortypes/nestedtensor/codegen/extension.py

```diff
@@ -0,0 +1,209 @@
+import torch
+from functools import wraps
+from collections import namedtuple
```

Comment:
A high level explanation of what this file does would be very helpful for the reviewer.